pages/_app.js

```diff
@@ -1,10 +1,45 @@
 import { GlobalStyle } from "@/GlobalStyles";
+import { useState } from "react";
+import { pets } from "@/Lib/Data";
+import { useRouter } from "next/router";
+import { uid } from "uid";
 
 export default function App({ Component, pageProps }) {
+  const [petCollection, setPetCollection] = useState(pets);
+  const router = useRouter();
+
+  function handleCreatePet(petData) {
+    setPetCollection((prevData) => [
+      {
+        id: uid(),
```

Comment:
Wouldn't it be easier to create the whole object already inside your form component? 

You accept the petData from your handleSubmit function and create the whole object inside that function.